File: iprep/Models/AppRoute.swift
import Foundation

public enum QuizMode: Hashable {
    case quickStart(resume: Bool)
    case module(id: String, title: String)
    case incorrectOnly
}

public enum AppRoute: Hashable {
    case dashboard
    case quiz(QuizMode)
    case progressReport
    case review
    case analytics
    case settings
}

File: shared/src/commonMain/kotlin/com/russellmiller/iprep/shared/models/AppRoute.kt
package com.russellmiller.iprep.shared.models

import kotlinx.serialization.Serializable

@Serializable
sealed class AppRoute {
    @Serializable
    object Dashboard : AppRoute()
    
    @Serializable
    data class Quiz(val mode: QuizMode) : AppRoute()
    
    @Serializable
    object ProgressReport : AppRoute()
    
    @Serializable
    object Review : AppRoute()
    
    @Serializable
    object Analytics : AppRoute()
    
    @Serializable
    object Settings : AppRoute()
    
    @Serializable
    object QuestionExplorer : AppRoute()
    
    @Serializable
    object Leaderboard : AppRoute()
}
